auth/server/src/main/java/com/seagates3/service/RequestorService.java

```diff
@@ -62,7 +64,26 @@ static Requestor getRequestor(ClientRequestToken clientRequestToken)
 
         try {
             perf.startClock();
-
+            if (AuthServerConfig.getCacheTimeout() != 0) {
+              GlobalData dataObj =
+                  GlobalDataStore.getInstance().getAuthenticationMap().get(
+                      clientRequestToken.getAccessKeyId());
+              if (dataObj != null) {
+                long difference =
+                    (System.currentTimeMillis() - dataObj.getCreationTime()) /
+                    1000;
+                LOGGER.debug("Cache time difference in seconds is - " +
+                             difference);
+                if (difference <
```

Comment:
check difference is not negative (!(difference < 0) && ( difference < AuthServerConfig.getCacheTimeout()))

This may happen if system time goes out of sync

Reply:
Fixed